api/types.go

```diff
@@ -138,10 +138,10 @@ type Timerange struct {
 
 // IsValid determines whether the given timerange meets the constraint.
 func (tr Timerange) IsValid() bool {
-	return (tr.Start%tr.Resolution == 0 &&
+	return tr.Resolution > 0 &&
+		tr.Start%tr.Resolution == 0 &&
 		tr.End%tr.Resolution == 0 &&
-		tr.Resolution > 0 &&
-		tr.Start <= tr.End)
+		tr.Start <= tr.End
 }
```

Comment:
I came across this when a panic happened while I was trying to fix up the tests to work with the new expression types.

Performing modulo by 0 causes a panic (like division by 0), so `tr.Resolution > 0` should be checked before the other cases.